tsl/test/expected/merge_chunks.out

```diff
@@ -0,0 +1,346 @@
+select "Constraint", "Columns", "Expr" from test.show_constraints('_timescaledb_internal._hyper_1_1_chunk');
+  Constraint  | Columns  |                                                                      Expr                                                                      
+--------------+----------+------------------------------------------------------------------------------------------------------------------------------------------------
+ constraint_1 | {time}   | (("time" >= 'Sun Dec 31 16:00:00 2023 PST'::timestamp with time zone) AND ("time" < 'Mon Jan 01 16:00:00 2024 PST'::timestamp with time zone))
+ constraint_2 | {device} | (_timescaledb_functions.get_partition_hash(device) < 715827882)
+(2 rows)
+
+-- Show partition layout
+select * from partitions;
+    table_name    | column_name |     range_start      |    range_end     
+------------------+-------------+----------------------+------------------
+ _hyper_1_1_chunk | time        |     1704067200000000 | 1704153600000000
+ _hyper_1_2_chunk | time        |     1704067200000000 | 1704153600000000
+ _hyper_1_1_chunk | device      | -9223372036854775808 |        715827882
+ _hyper_1_2_chunk | device      |            715827882 |       1431655764
+(4 rows)
+
+-- Now merge chunk 1 and 2:
+begin;
+call merge_chunks('_timescaledb_internal._hyper_1_1_chunk', '_timescaledb_internal._hyper_1_2_chunk');
+select * from _timescaledb_internal._hyper_1_1_chunk;
+             time             | device | temp 
+------------------------------+--------+------
+ Mon Jan 01 00:00:00 2024 PST |      1 |    1
+ Mon Jan 01 00:00:00 2024 PST |      2 |    2
+(2 rows)
+
+select reltuples from pg_class where oid='_timescaledb_internal._hyper_1_1_chunk'::regclass;
+ reltuples 
+-----------
+         2
+(1 row)
+
+select * from partitions;
+    table_name    | column_name |     range_start      |    range_end     
+------------------+-------------+----------------------+------------------
+ _hyper_1_1_chunk | time        |     1704067200000000 | 1704153600000000
+ _hyper_1_1_chunk | device      | -9223372036854775808 |       1431655764
+(2 rows)
+
+select "Constraint", "Columns", "Expr" from test.show_constraints('_timescaledb_internal._hyper_1_1_chunk');
+  Constraint  | Columns  |                                                                      Expr                                                                      
+--------------+----------+------------------------------------------------------------------------------------------------------------------------------------------------
+ constraint_1 | {time}   | (("time" >= 'Sun Dec 31 16:00:00 2023 PST'::timestamp with time zone) AND ("time" < 'Mon Jan 01 16:00:00 2024 PST'::timestamp with time zone))
+ constraint_2 | {device} | (_timescaledb_functions.get_partition_hash(device) < 1431655764)
+(2 rows)
+
+select count(*) as num_orphaned_slices from orphaned_slices;
+ num_orphaned_slices 
+---------------------
+                   0
```

Comment:
Are there a way to have orphaned slices other than manually mess with catalog?

Reply:
Since the merge code is messing with the slices to redefine partition boundaries, I wanted to make sure it works correctly. Slices are shared between chunks and when you delete chunks and redefine slice boundaries, you want to make sure a slice isn't orphaned.

There's a check to clean up orphaned slices in `chunk_update_constraint` (part of this PR), so this effectively tests that it is doing its job.